src/mscorlib/src/System/Reflection/RuntimeMethodInfo.cs

```diff
@@ -41,7 +41,7 @@ internal INVOCATION_FLAGS InvocationFlags
                     //
                     // first take care of all the NO_INVOKE cases. 
                     if (ContainsGenericParameters ||
-                         ReturnType.IsByRef ||
+                         (ReturnType.IsByRef && ReturnType.GetElementType().IsByRefLike) ||
```

Comment:
Update the message that will be thrown for this case in NotSupported_ByRefReturn ?

Reply:
Fixed.